lib/src/lifecycle_module.dart

```diff
@@ -147,24 +189,41 @@ abstract class LifecycleModule extends Object with Disposable {
   /// Calls the onLoad() method, which can be implemented on a Module.
   /// Executes the willLoad and didLoad event streams.
   ///
-  /// [LifecycleModule] only supports one load/unload cycle. If [load] is called
-  /// again after a module has been unloaded, a [StateError] is thrown.
+  /// Initiates the loading process when the module is in the initialized state.
+  /// If the module is in the loaded or loading state a warning is logged and
+  /// the method is a noop. If the module is in any other state, a StateError is
+  /// thrown.
+  ///
+  /// Note that [LifecycleModule] only supports one load/unload cycle. If [load]
+  /// is called after a module has been unloaded, a [StateError] is thrown.
   Future<Null> load() {
-    final completer = new Completer<Null>();
-    if (_isUnloadedOrUnloading) {
-      completer
-          .completeError(new StateError('Module "$name" cannot be reloaded.'));
-    } else if (!_isLoaded) {
-      _willLoadController.add(this);
-      onLoad().then((_) {
-        _didLoadController.add(this);
-        _isLoaded = true;
-        completer.complete();
-      });
-    } else {
+    if (isLoading) {
+      _logger.warning('Module "$name" is already loading, cannot load.');
+      return _transitionFuture;
+    }
+
+    if (isLoaded) {
       _logger.warning('Module "$name" is already loaded, cannot load.');
-      completer.complete();
+      return new Future.value(null);
+    }
```

Comment:
Would it be worth merging this block a bit with the `isLoading` block to DRY up the logging message?

```dart
if (isLoading || isLoaded) {
  _logger.warning(...);
  if (isLoading) return _transitionFuture;
  else return new Future.value(null);
}
```

I don't have a strong opinion on this so if you'd prefer to keep it how it is for clarity, 👍